src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExporter.cs

```diff
@@ -62,75 +63,123 @@ internal DynatraceMetricsExporter(DynatraceExporterOptions options, ILogger<Dyna
+						if (response.IsSuccessStatusCode)
+						{
+							exportResult = ExportResult.Success;
+						}
+						else
+						{
+							_logger.ReceivedErrorResponse(
+								response.StatusCode, response.Content.ReadAsStringAsync().GetAwaiter().GetResult());
+							exportResult = ExportResult.Failure;
+						}
+					}
+					catch (Exception ex)
+					{
+						_logger.FailedSendingMetricLines(ex);
+						exportResult = ExportResult.Failure;
+					}
+
+					if (exportResult == ExportResult.Failure)
+					{
+						// if any batch fails, the entire batch should fail.
+						return exportResult;
 					}
 				}
-				catch (Exception e)
-				{
-					_logger.LogError("Error sending metrics: {Error}", e.Message);
-					throw;
-				}
+
+				return ExportResult.Success;
 			}
+		}
 
-			// if all chunks were exported successfully, return success, otherwise failed.
-			return exportResults.All(x => x == ExportResult.Success) ? ExportResult.Success : ExportResult.FailedNotRetryable;
+		private IReadOnlyCollection<string> GetSerializeMetricLines(in Batch<Metric> batch)
+		{
+			var metricLines = new List<string>();
+
+			foreach (var metric in batch)
+			{
+				metricLines.AddRange(GetSerializedMetricPoints(metric));
+			}
+			return metricLines;
 		}
 
-		private void SerializeMetric(StringBuilder sb, Metric metric)
+		private IReadOnlyCollection<string> GetSerializedMetricPoints(Metric metric)
 		{
-			foreach (var dynatraceMetric in metric.ToDynatraceMetrics(_logger))
+			// TODO: Try to send a PR to get a .Count of the metric points
+			// so we can allocate only the necessary in the list. Similar as in:
+			// https://github.com/open-telemetry/opentelemetry-dotnet/pull/2542/files
+
+			var lines = new List<string>();
+			foreach (var metricPoint in metric.GetMetricPoints())
 			{
+				DynatraceMetric dtMetric = null;
 				try
 				{
-					sb.AppendLine(_serializer.SerializeMetric(dynatraceMetric));
+					switch (metric.MetricType)
+					{
+						case MetricType.LongSum:
+							dtMetric = metric.ToLongCounterDelta(metricPoint, _logger);
+							break;
+						case MetricType.DoubleSum:
+							dtMetric = metric.ToDoubleCounterDelta(metricPoint, _logger);
+							break;
+						case MetricType.LongGauge:
+							dtMetric = metric.ToLongGauge(metricPoint, _logger);
+							break;
+						case MetricType.DoubleGauge:
+							dtMetric = metric.ToDoubleGauge(metricPoint, _logger);
+							break;
+						case MetricType.Histogram:
+							dtMetric = metric.ToDoubleHistogram(metricPoint, _logger);
+							break;
+						default:
+							_logger.InvalidMetricType(metric.MetricType);
+							break;
+					}
+					if (dtMetric != null)
+					{
+						lines.Add(_serializer.SerializeMetric(dtMetric));
+					}
+				}
+				catch (DynatraceMetricException dtEx)
+				{
+					_logger.FailedToSerializeMetric(metric.Name, dtEx);
 				}
-				catch (DynatraceMetricException e)
+				catch (Exception ex)
```

Comment:
Since this does the same thing we could combine it, right? :thinking:

Reply:
True. Not sure why I had both. Kept only `Exception`.